plugins/DebuggerCore/unix/linux/arch/x86-generic/PlatformState.cpp

```diff
@@ -478,6 +481,9 @@ bool PlatformState::fillFrom(const X86XState &regs, size_t sizeFromKernel) {
 		// Only fill the registers which are actually supported, leave invalidity marks intact for other parts
 		if (avx.xcr0 & X86XState::FEATURE_AVX) { // If AVX state management has been enabled by the OS
 			for (size_t n = 0; n < MAX_YMM_REG_COUNT; ++n) {
+
+				// TODO(eteran): a default initialized value should just have all zeros set
+				// no need to use fromZeroExtended, right?
```

Comment:
When I was writing all this stuff, I supposed that default values would be `bad1bad1bad1....` Zero is a very bad choice to default-initialize state data in EDB, since a bug in treating unfilled state/registers will then be hidden. And we have actually had such a bug — when I started making my first commits here.

Reply:
Very good point. Glad that I made this a comment, and didn't just change it :-)